src/controllers/simplified-account/services/payment-links/create/payment-link-information.controller.test.ts

```diff
@@ -0,0 +1,421 @@
+import ControllerTestBuilder from '@test/test-helpers/simplified-account/controllers/ControllerTestBuilder.class'
+import sinon from 'sinon'
+import GatewayAccountType from '@models/gateway-account/gateway-account-type'
+import formatServiceAndAccountPathsFor from '@utils/simplified-account/format/format-service-and-account-paths-for'
+import formatAccountPathsFor from '@utils/format-account-paths-for'
+import paths from '@root/paths'
+import { expect } from 'chai'
+
+interface SessionWithPageData {
+  pageData?: {
+    createPaymentLink?: {
+      paymentLinkTitle?: string
+      paymentLinkDescription?: string
+      serviceNamePath?: string
+      productNamePath?: string
+      isWelsh?: boolean
+      gatewayAccountId?: number
+      paymentLinkAmount?: number
+      paymentReferenceType?: string
+      paymentReferenceLabel?: string
+      paymentReferenceHint?: string
+      amountHint?: string
+    }
+  }
+}
+
+const SERVICE_EXTERNAL_ID = 'service123abc'
+const GATEWAY_ACCOUNT_ID = 117
+const GATEWAY_ACCOUNT_EXTERNAL_ID = 'gateway-account-external-id-123'
+
+const mockResponse = sinon.spy()
+
+const mockNunjucksFilters = {
+  slugify: sinon.stub().callsFake((str: string) => str.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-+|-+$/g, '')),
+  removeIndefiniteArticles: sinon.stub().callsFake((str: string) => str.replace(/^(a|an|the)\s+/i, ''))
```

Comment:
do these need to be stubbed? they don't make network calls so stubbing just adds needless complexity to the tests